cwf/cloud/go/services/cwf/analytics/analytics.go

```diff
@@ -69,58 +68,59 @@ func GetAnalyticsCalculations(config *cwf_service.Config) []calculations.Calcula
-	allCalculations = append(allCalculations, getXAPCalculations(daysToCalculate, xapGauge, activeUsersMetricName, metricConfig)...)
+	allCalculations = append(allCalculations, getXAPCalculations(daysToCalculate, xapGauge, activeUsersMetricName)...)
 
 	// User Throughput Calculations
-	allCalculations = append(allCalculations, getUserThroughputCalculations(daysToCalculate, userThroughputGauge, userThroughputMetricName, metricConfig)...)
+	allCalculations = append(allCalculations, getUserThroughputCalculations(daysToCalculate, userThroughputGauge, userThroughputMetricName)...)
 
 	// AP Throughput Calculations
-	allCalculations = append(allCalculations, getAPNThroughputCalculations(daysToCalculate, apThroughputGauge, apThroughputMetricName, metricConfig)...)
+	allCalculations = append(allCalculations, getAPNThroughputCalculations(daysToCalculate, apThroughputGauge, apThroughputMetricName)...)
 
 	// User Consumption Calculations
-	allCalculations = append(allCalculations, getUserConsumptionCalculations(daysToCalculate, userConsumptionGauge, userConsumptionMetricName, metricConfig)...)
-	allCalculations = append(allCalculations, get1hourConsumptionCalculation(hourlyUserConsumptionGauge, userConsumptionHourlyMetricName, metricConfig)...)
+	allCalculations = append(allCalculations, getUserConsumptionCalculations(daysToCalculate, userConsumptionGauge, userConsumptionMetricName)...)
+	allCalculations = append(allCalculations, get1hourConsumptionCalculation(hourlyUserConsumptionGauge, userConsumptionHourlyMetricName)...)
 
 	// Authentication Calculations
-	allCalculations = append(allCalculations, getAuthenticationCalculations(daysToCalculate, authenticationsGauge, authenticationsMetricName, metricConfig)...)
+	allCalculations = append(allCalculations, getAuthenticationCalculations(daysToCalculate, authenticationsGauge, authenticationsMetricName)...)
 
 	// Raw Metrics
-	allCalculations = append(allCalculations, calculations.GetRawMetricsCalculations(metricConfig)...)
+	allCalculations = append(allCalculations, calculations.GetRawMetricsCalculations(analyticsConfig)...)
 
 	return allCalculations
 }
 
-func getXAPCalculations(daysList []int, gauge *prometheus.GaugeVec, metricName string, metricConfig map[string]calculations.MetricConfig) []calculations.Calculation {
+func getXAPCalculations(daysList []int, gauge *prometheus.GaugeVec, metricName string) []calculations.Calculation {
 	calcs := make([]calculations.Calculation, 0)
 	for _, dayParam := range daysList {
 		calcs = append(calcs, &cwf_calculations.XAPCalculation{
-			CalculationParams: calculations.CalculationParams{
-				Days:                dayParam,
-				RegisteredGauge:     gauge,
-				Labels:              prometheus.Labels{calculations.DaysLabel: strconv.Itoa(dayParam)},
-				Name:                metricName,
-				ExpectedGaugeLabels: xapLabels,
-				MetricConfig:        metricConfig,
+			BaseCalculation: calculations.BaseCalculation{
+				CalculationParams: &calculations.CalculationParams{
```

Comment:
Here and elsewhere: I'm not sure but I feel relatively certain there has to be a less-nested way to handle these calculations